com.avaloq.tools.ddk.xtext.builder/src/com/avaloq/tools/ddk/xtext/builder/MonitoredClusteringBuilderState.java

```diff
@@ -699,7 +712,12 @@ protected void storeBinaryResource(final Resource resource, final BuildData buil
         && fileSystemAccess instanceof IFileSystemAccessExtension3) {
 
       try {
-        CompletableFuture.runAsync(() -> doStoreBinaryResource(resource, buildData), binaryStorageExecutor);
+        int currentQueueSize = binaryStorageExecutor.getQueue().size();
+        if (currentQueueSize > binaryStorageHighWaterMark) {

```

Comment:
what about recording this only if the queue is big enough? Let us say, > 1_000?

Reply:
we could, but I don't see why we would want to